### ISSUE TITLE
TOC - displays  invalid heading from example markdown code

### ISSUE THREAD

**Issue by stefansimik** [jupyterlab/jupyterlab#13948]
When you put example HTML code inside Markdown cell:

```html
<!DOCTYPE html>
<html>
  <body>
    **<h1>Heading element</h1>**
    <p>Paragraph element</p>
  </body>
</html>
```

Then real heading of type H1 is displayed, which should not happen, because it is just example of code.

Jupyterlab version: 3.4.4

See: 

![Screenshot 2023-02-08 at 22 21 55](https://user-images.githubusercontent.com/6208056/217655730-e485f1e9-5773-4204-922e-9b6e3b6edc8b.png)

### Solution
 
There should be very simple and consistent rule.
Anything, that is wrapped in code quotes, should be ignored, i.e

Anything wrapped in single backticks like this:
`code like this`

or triple backticks like this:

```python
x=1
print(x)
```

**Comment by JasonWeill** [comment]
@stefansimik Thanks for filing this! Do you still see this problem on newer versions of JupyterLab? You can try them out via Binder using links here: https://gist.github.com/jtpio/c4b6376bfbd00474b3c0d1d60fe23670

**Comment by JasonWeill** [comment]
After testing in 4.0.0 with the rewritten Table of Contents (TOC) extension, I can verify that this bug is fixed.

![image](https://user-images.githubusercontent.com/93281816/217904852-d31a3d83-4793-4e30-9dea-d1310b44dbd7.png)

**Comment by JasonWeill** [comment]
Reopened for consideration of a fix in the 3.6.x series.

**Comment by stefansimik** [comment]
Excellent work, thank you very much ❤️
